zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala

```diff
@@ -0,0 +1,87 @@
+package zio.cache
+
+import zio.{Exit, IO, Managed, TaskManaged, UIO, URIO}
+
+import java.time.Duration
+
+
+sealed abstract class ManagedCache[-Key, +Error, +Value] {
+
+  /**
+   * Returns statistics for this cache.
+   * @return
+   */
+  def cacheStats: UIO[CacheStats]
+
+  /**
```

Comment:
Yes, I think since it is an asynchronous cache the cache contains a key if either there is an existing value in the cache or someone is in the process of putting that value in the cache.